Dockerfile

```diff
@@ -18,117 +20,129 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     curl \
     kdiff3 \
     git \
-    vim # for debug
+    vim \
+  && rm -rf /var/lib/apt/lists/*
 
 # Install runtime packages.
-RUN apt-get update && \
-    apt-get install -y \
-    ruby \
-    python \
-    python-dev \
-    python-pip \
+RUN apt-get update \
+  && apt-get install -y \
     unzip \
     perl \
     software-properties-common \
-    php-pear
+    php-pear \
+    python3-pip \
+  && rm -rf /var/lib/apt/lists/*
+
+# Install protoc 3.3.0.
+# Currently, the stable version is still 2.6.x, which can only handle proto2
+# syntax, so we have to download our own.
+RUN mkdir -p /usr/src/protoc/ \
+  && curl --location https://github.com/google/protobuf/releases/download/v3.3.0/protoc-3.3.0-linux-x86_64.zip > /usr/src/protoc/protoc-3.3.0.zip \
+  && cd /usr/src/protoc/ \
+  && unzip protoc-3.3.0.zip \
+  && ln -s /usr/src/protoc/bin/protoc /usr/local/bin/protoc
+
+# Install GRPC and Protobuf.
+RUN pip3 install --upgrade pip \
+  && pip3 install \
+    grpcio==1.3.0 \
+    protobuf==3.3.0
 
 # Install Oracle JDK 8
-RUN add-apt-repository ppa:openjdk-r/ppa
-RUN apt-get update && apt-get install -y openjdk-8-jdk
+RUN add-apt-repository ppa:openjdk-r/ppa \
+  && apt-get update \
+  && apt-get install -y openjdk-8-jdk \
+  && rm -rf /var/lib/apt/lists/*
 
 # Setup JAVA_HOME, this is useful for docker commandline
 ENV JAVA_HOME /usr/lib/jvm/java-8-openjdk-amd64/
 RUN export JAVA_HOME
 
-# Install NodeJS
-ADD ./scripts /scripts
-WORKDIR /scripts
-# This is need to install nodejs 4.x otherwise nodejs 0.x will be installed.
-RUN bash setup_node4.sh
-RUN apt-get install -y nodejs
-
-# Install linuxbrew.
-RUN useradd -m -s /bin/bash linuxbrew
```

Comment:
I was able to completely nix Linuxbrew because Ubuntu 16.04 has a new enough version of everything, meaning we no longer incur the overhead of an entirely new package manager. This saved about 200 MB.